cpp/src/neighbors/detail/nn_descent.cuh

```diff
@@ -44,12 +50,13 @@
 #include <omp.h>
 
 #include <limits>
+#include <optional>
 #include <queue>
 #include <random>
 
 namespace cuvs::neighbors::nn_descent::detail {
-static const std::string RAFT_NAME = "raft";
-using pinned_memory_resource       = thrust::universal_host_pinned_memory_resource;
+
+using pinned_memory_resource = thrust::universal_host_pinned_memory_resource;
```

Comment:
We should work to minimize the direct thrust calls in this algorithm. Not somehting that needs to be done in this PR, but can you create an issue to do this as tech debt in some future version? 

The more direct calls we make to libraries we don't control, the more things can break as those libraries evolve. We have a lot of pinned stuff that we do within RAFT, and we should try to use RAFT calls as much as possible.